File: 415-add-strings/add-strings.go
func addStrings(num1 string, num2 string) string {
    res := []byte{}
	i, j := len(num1)-1, len(num2)-1
	carry := 0

	for i >= 0 || j >= 0 || carry > 0 {
		sum := carry
		if i >= 0 {
			sum += int(num1[i] - '0')
			i--
		}
        
		if j >= 0 {
			sum += int(num2[j] - '0')
			j--
		}

		res = append(res, byte(sum%10)+'0')
		carry = sum / 10
	}

	for k, l := 0, len(res)-1; k < l; k, l = k+1, l-1 {
		res[k], res[l] = res[l], res[k]
	}

	return string(res)
}

File: 415-add-strings/add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        res = []
        carry = 0
        
        p1 = len(num1) - 1
        p2 = len(num2) - 1
        
        while p1 >= 0 or p2 >= 0 or carry:
            digit1 = ord(num1[p1]) - ord('0') if p1 >= 0 else 0
            digit2 = ord(num2[p2]) - ord('0') if p2 >= 0 else 0
            
            current_sum = digit1 + digit2 + carry
            carry = current_sum // 10
            res.append(str(current_sum % 10))
            
            p1 -= 1
            p2 -= 1
            
        return "".join(res[::-1])